runtime/criusupport/criusupport.c

```diff
@@ -120,80 +106,100 @@ Java_org_eclipse_openj9_criu_CRIUSupport_isCheckpointAllowed(JNIEnv *env, jclass
+#define J9_NATIVE_STRING_FAIL_TO_CONVERT (-2)
+
+/**
+ * Converts the given java string to native representation. The nativeString parameter should point
+ * to a buffer with its size specified by nativeStringBufSize. If the java string length exceeds the buffer
+ * size, nativeString will be set to allocated memory.
+ *
+ * @note If successful, the caller is responsible for freeing any memory allocated and stored in nativeString.
+ *
+ * @param[in] currentThread current thread
+ * @param[in] javaString java string object
+ * @param[out] nativeString the native representation of the java string
+ * @param[in] nativeStringBufSize size of the nativeString buffer
+ *
+ * @return return code indicating success, allocation failure, or string conversion failure
+ */
+static IDATA
+getNativeString(J9VMThread *currentThread, j9object_t javaString, char **nativeString, IDATA nativeStringBufSize)
 {
 	J9JavaVM *vm = currentThread->javaVM;
 	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
 	char mutf8StringBuf[STRING_BUFFER_SIZE];
 	char *mutf8String = NULL;
 	UDATA mutf8StringSize = 0;
-	UDATA requiredConvertedStringSize = 0;
-	BOOLEAN res = TRUE;
+	IDATA requiredConvertedStringSize = 0;
+	char *localNativeString = *nativeString;
+	IDATA res = J9_NATIVE_STRING_NO_ERROR;
 	PORT_ACCESS_FROM_VMC(currentThread);
 	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
 
 	mutf8String = vmFuncs->copyStringToUTF8WithMemAlloc(currentThread, javaString, J9_STR_NULL_TERMINATE_RESULT, "", 0, mutf8StringBuf, STRING_BUFFER_SIZE, &mutf8StringSize);
 	if (NULL == mutf8String) {
-		vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
-		*resultType = vm->criuSupportJVMCheckpointFailure;
-		res = FALSE;
+		res = J9_NATIVE_STRING_OUT_OF_MEMORY;
 		goto free;
 	}
 
 	/* get the required size */
 	requiredConvertedStringSize = omrstr_convert(J9STR_CODE_MUTF8, J9STR_CODE_PLATFORM_RAW,
 					mutf8String,
 					mutf8StringSize,
-					*nativeString,
+					localNativeString,
 					0);
 
 	if (requiredConvertedStringSize < 0) {
-		vmFuncs->setCurrentExceptionNLSWithArgs(currentThread, J9NLS_JCL_CRIU_FAILED_TO_CONVERT_JAVA_STRING, J9VMCONSTANTPOOL_JAVALANGINTERNALERROR, requiredConvertedStringSize);
-		*resultType = vm->criuSupportInvalidArguments;
-		res = FALSE;
+		Trc_CRIU_getNativeString_getStringSizeFail(currentThread, mutf8String, mutf8StringSize);
+		res = J9_NATIVE_STRING_FAIL_TO_CONVERT;
 		goto free;
 	}
 
 	/* Add 1 for NUL terminator */
 	requiredConvertedStringSize += 1;
 
-	if (requiredConvertedStringSize > STRING_BUFFER_SIZE) {
-		*nativeString = (char*)j9mem_allocate_memory(requiredConvertedStringSize, OMRMEM_CATEGORY_VM);
-		if (NULL == *nativeString) {
-			vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
-			*resultType = vm->criuSupportJVMCheckpointFailure;
-			res = FALSE;
+	if (requiredConvertedStringSize > nativeStringBufSize) {
+		localNativeString = (char*)j9mem_allocate_memory(requiredConvertedStringSize, OMRMEM_CATEGORY_VM);
```

Comment:
In C, it's not necessary to cast a void pointer.  It autoconverts
```suggestion
		localNativeString = j9mem_allocate_memory(requiredConvertedStringSize, OMRMEM_CATEGORY_VM);
```